src/index.ts

```diff
@@ -1,19 +1,649 @@
-export * from "./lib/mapping/getActiveAddress";
-export * from "./lib/mapping/getActivePublicKey";
-export * from "./lib/mapping/getWalletNames";
-export * from "./lib/mapping/sign";
-export * from "./lib/mapping/signature";
-export * from "./lib/mapping/dispatch";
-export * from "./lib/mapping/signMessage";
-export * from "./lib/mapping/verifyMessage";
-export * from "./lib/mapping/signDataItem";
-// types
-export * from "./types/mapping/connect";
-
 import { Buffer } from "buffer";
+import { OthentAuth0Client } from "./lib/auth/auth0";
+import Transaction, { Tag } from "arweave/web/lib/transaction";
+import { addOthentAnalyticsTags } from "./lib/analytics/analytics.utils";
+import type Arweave from "arweave/web";
+import {
+  BinaryDataType,
+  binaryDataTypeOrStringToBinaryDataType,
+  bufferTob64Url,
+  hash,
+  stringToUint8Array,
+} from "./lib/utils/arweaveUtils";
+import { createData, DataItemCreateOptions, Signer } from "warp-arbundles";
+import { OthentKMSClient } from "./lib/othent-kms-client/client";
+import {
+  Auth0Strategy,
+  UserDetails,
+} from "./lib/auth/auth0.types";
+import {
+  ArConnect,
+  DataItem,
+  DispatchResult,
+  PermissionType,
+  SignMessageOptions,
+} from "./types/arconnect/arconnect.types";
+import {
+  ANALYTICS_TAGS,
+  CLIENT_NAME,
+  CLIENT_VERSION,
+  DEFAULT_OTHENT_CONFIG,
+} from "./lib/config/config.constants";
+
+// Type exports:
+
+export {
+  DispatchResult,
+  SignMessageOptions,
+} from "./types/arconnect/arconnect.types";
+export { TypedArray, BinaryDataType } from "./lib/utils/arweaveUtils";
+
+export {
+  Auth0Strategy,
+  IdTokenWithData,
+  UserDetails,
+} from "./lib/auth/auth0.types";
+
+// Constant exports:
+
+export {
+  DEFAULT_OTHENT_CONFIG,
+  CLIENT_NAME,
+  CLIENT_VERSION,
+} from "./lib/config/config.constants";
+
+// TODO: Polyfill. Should we overwrite a global like this from a library?
+
 window.Buffer = Buffer;
+
+export interface OthentConfig {
+  auth0Domain: string;
+  auth0ClientId: string;
+  auth0Strategy: Auth0Strategy;
+  serverBaseURL: string;
+}
+
+export interface OthentOptions extends Partial<OthentConfig> {
+  crypto?: Crypto | null;
+}
+
+export class Othent
+  implements Omit<ArConnect, "connect" | "dispatch" | "signDataItem">
+{
+  walletName = CLIENT_NAME;
+
+  walletVersion = CLIENT_VERSION;
+
+  config: OthentConfig = DEFAULT_OTHENT_CONFIG;
+
+  crypto: Crypto;
+
+  api: OthentKMSClient;
+
+  auth0: OthentAuth0Client;
+
+  // TODO: Add listener for user details change?
+
+  // TODO: Add listener for errors and a silentErrors: boolean property?
+
+  // TODO: When using refresh tokens in memory, the developer has to manually call connect() before calling any other function, as otherwise
+  // get a "Missing cached user." error. Can we improve that?
+
+  constructor(options: OthentOptions = {}) {
+    let { crypto: cryptoOption, ...configOptions } = options;
+
+    this.config = { ...configOptions, ...DEFAULT_OTHENT_CONFIG };
+
+    let crypto = cryptoOption;
```

Comment:
@7i7o Crypto module config. This should work in all environments that have a Crypto module (e.g. Node.js) or can provide it (e.g. React Native). It might be worth testing everything actually runs fine in practice with some automated tests (eventually).

Reply:
Great addition. As we found out, some of our dependencies also use `crypto` under the hood and that might force devs to also polyfill the package anyway... wdyt?

Reply:
Yep, because of that, I've removed the `crypto` option and just left some logic to check if that API exists. If it does not, an error is thrown letting developers know they need to provide a polyfill. Because `arweave` also uses it internally, it didn't make sense to keep the option.